homeassistant/components/smappee/config_flow.py

```diff
@@ -21,3 +27,69 @@ class SmappeeFlowHandler(
     def logger(self) -> logging.Logger:
         """Return logger."""
         return logging.getLogger(__name__)
+
+    async def async_step_zeroconf(self, discovery_info=None):
+        """Handle zeroconf discovery."""
+        if discovery_info is None:
+            return self.async_abort(reason="connection_error")
+
+        if not discovery_info[CONF_HOSTNAME].startswith("Smappee1"):
+            # We currently only support Energy and Solar models (legacy)
+            return self.async_abort(reason="invalid mDNS")
+
+        # pylint: E1101: Instance of 'SmappeeFlowHandler' has no 'context' member (no-member)
+        self.context = {"source": "zeroconf"}
```

Comment:
We don't need to create the context just update it. It should already have the correct source.

Reply:
The tests seem to fail if I don't create the context

Reply:
You need to set/mock the context in the tests.
Example: <https://github.com/home-assistant/core/blob/dev/tests/components/wled/test_config_flow.py#L155>

Reply:
The context has been set in the tests (async_init methods) but the error still occurs. What am I not seeing?